src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.h

```diff
@@ -155,49 +201,67 @@ struct BallPlacementPlayFSM
 
         DEFINE_SML_ACTION(alignPlacement)
         DEFINE_SML_ACTION(placeBall)
-        DEFINE_SML_ACTION(kickOffWall)
+        DEFINE_SML_ACTION(setPickOffDest)
+        DEFINE_SML_ACTION(alignWall)
+        DEFINE_SML_ACTION(pickOffWall)
+        DEFINE_SML_ACTION(releaseBall)
         DEFINE_SML_ACTION(startWait)
         DEFINE_SML_ACTION(retreat)
+        DEFINE_SML_ACTION(wait)
 
-        DEFINE_SML_GUARD(shouldKickOffWall)
+        DEFINE_SML_GUARD(shouldPickOffWall)
         DEFINE_SML_GUARD(alignDone)
-        DEFINE_SML_GUARD(kickDone)
+        DEFINE_SML_GUARD(wallAlignDone)
+        DEFINE_SML_GUARD(wallPickOffDone)
+        DEFINE_SML_GUARD(ballReleased)
         DEFINE_SML_GUARD(ballPlaced)
         DEFINE_SML_GUARD(waitDone)
         DEFINE_SML_GUARD(retreatDone)
 
         return make_transition_table(
```

Comment:
Could you add an empty line between lines where the states are change to improve readability of the long FSM? Could also include some comments to clarify what each state does.